app/components/TransactionModal.tsx

```diff
@@ -0,0 +1,520 @@
+                            aria-label="Close modal"
+                            onClick={onClose}
+                            className="rounded-lg p-2 hover:bg-gray-100 dark:hover:bg-white/10 max-sm:-ml-2 sm:hidden"
+                          >
+                            <ArrowLeft02Icon className="size-5 text-outline-gray dark:text-white/50" />
+                          </button>
+                          <h2 className="text-lg font-semibold text-text-body dark:text-white sm:flex-1">
+                            Transactions
+                          </h2>
+                          <button
+                            type="button"
+                            title="Close"
+                            onClick={onClose}
+                            className="rounded-lg p-2 hover:bg-gray-100 dark:hover:bg-white/10"
+                          >
+                            <Cancel01Icon className="size-5 text-outline-gray dark:text-white/50" />
+                          </button>
+                        </div>
+
+                        <div className="max-h-[80vh] overflow-y-auto pb-4 scrollbar-hide">
+                          {transactions.map((group) => (
+                            <div key={group.id} className="space-y-3">
+                              <div className="flex items-center justify-between gap-x-6">
+                                <h3 className="text-sm font-medium text-text-secondary dark:text-white/50">
+                                  {group.date}
+                                </h3>
+                                <div className="w-2/3 border border-dashed border-[#EBEBEF] dark:border-[#FFFFFF1A]" />
+                              </div>
+                              <div className="space-y-2">
+                              {group.items.map((transaction) => (
+                                  <div
+                                    key={transaction.id}
+                                    onClick={() =>
+                                      setSelectedTransaction(transaction)
+                                    }
+                                    className="flex cursor-pointer items-center justify-between py-2"
+                                  >
+                                    {/* Transaction row content */}
+                                    <div className="flex items-center gap-2">
+                                      {/* Icon based on currency */}
+                                      <div className="text-sm">
+                                        <div className="flex items-center gap-x-2">
+                                          <Image
+                                            src={
+                                              transaction?.currency === "USDC"
+                                                ? "/logos/usdc-logo.svg"
+                                                : "/logos/usdt-logo.svg"
+                                            }
```

Comment:
cNGN is supported, but this doesn't handle if cNGN is the tx token.

based on the naming pattern in the codebase, and you can simply just do 

```tsx
{`/logos/${String(token)?.toLowerCase()}-logo.svg`}
```

ref: `app\pages\TransactionPreview.tsx`, line 450